# mirna_pipeline
--------------------------------------
Bioinformatic pipeline to analyze micro RNA sequencing data

## Differential expression analysis

Even though a small proportion of reads was mapped to human miRNA,
some experimental results confirming the bioinformatic predictions
convinced us that reads mapped to miRNA are true miRNA reads, and
we proceeded with the analysis of differentially expressed microRNAs.

**Input file generated by the core**

Input *xls* files containing numbers of reads mapped to various human
miRNAs were merged into a 
[matrix](https://github.com/jknightlab/mirna_pipeline/edit/master/mirna_matrix_no_285_DESeq_intput.txt).
This matrix was used as an input file for `DESeq` -- the tool used to generate
lists of differentially expressed miRNAs.

| Sample name | Sample type |
| ----------- | ----------- |
| 274         | HC1-nonTh17 |
| 276         | HC2-nonTh17 |
| 278         | HC3-nonTh17 |
| 280         | HC4-nonTh17 |
|             |             |
| 275         | HC1-Th17    |
| 277         | HC2-Th17    |
| 279         | HC3-Th17    |
| 281         | HC4-Th17    |
|             |             |
| 282         | AS1-nonTh17 |
| 284         | AS2-nonTh17 |
| 286         | AS3-nonTh17 |
| 288         | AS4-nonTh17 |
|             |             |
| 283         | AS1-Th17    |
| 289         | AS4-Th17    |


**Differentially expressed miRNAs with DESeq**

Running [`DESeq`](https://bioconductor.org/packages/release/bioc/html/DESeq.html)
was performed following some online 
[guidelines](http://dwheelerau.com/2013/04/15/how-to-use-deseq-to-analyse-rnaseq-data/).

On `rescomp1`:

`> module load R/3.1.2`
`R`

In `R`:

```
library("DESeq")
countsTable <- read.delim("input_deseq_mirna_matrix.txt", header=TRUE)
rownames(countsTable) <- countsTable$mirna
countsTable <- countsTable[,-1]
conds <- factor(c("HC_Th17", "HC_nonTh17", "HC_Th17", "HC_nonTh17", "HC_Th17", "HC_nonTh17", "HC_Th17", "AS_nonTh17", "AS_Th17", "AS_nonTh17", "AS_Th17", "AS_nonTh17", "AS_nonTh17"))
cds <- newCountDataSet(countsTable, conds)
cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)
```

Creating a plot to estimate dispersion:
```
plotDispEsts <- function(cds){
 plot(rowMeans(counts(cds, normalized=TRUE)),
 fitInfo(cds)$perGeneDispEsts,
 pch='.', log="xy",
 xlab="mean of normalized counts", ylab="dispersion",
 )
 xg = 10^seq( -.5, 5, length.out=300 )
 lines(xg, fitInfo(cds)$dispFun(xg), col="red")
}
png("dispersion_estimation.png")
plotDispEsts(cds)
dev.off()
```

![alt text](https://github.com/jknightlab/mirna_pipeline/blob/master/dispersion_estimation.png)

Measuring actual differential expression:
```
# healthy control, differences between Th17 and non-Th17
res <- nbinomTest(cds, "HC_Th17", "HC_nonTh17")
png("log_fold_change_HC_Th17_VS_HC_nonTh17.png")
plotMA(res)
dev.off()
write.table(res, file="diff_expression_HC_Th17_VS_HC_nonTh17.txt", quote=FALSE, sep="\t", row.names=FALSE)
```

More code (for all comparisons):
```
res <- nbinomTest(cds, "HC_Th17", "AS_Th17")
write.table(res, file="diff_expression_HC_Th17_VS_AS_Th17.txt", quote=FALSE, sep="\t", row.names=FALSE)
# ===============================================
res <- nbinomTest(cds, "HC_Th17", "AS_nonTh17")
write.table(res, file="diff_expression_HC_Th17_VS_AS_nonTh17.txt", quote=FALSE, sep="\t", row.names=FALSE)
# ===============================================
res <- nbinomTest(cds, "HC_nonTh17", "AS_Th17")
write.table(res, file="diff_expression_HC_nonTh17_VS_AS_Th17.txt", quote=FALSE, sep="\t", row.names=FALSE)
# ===============================================
res <- nbinomTest(cds, "HC_nonTh17", "AS_nonTh17")
write.table(res, file="diff_expression_HC_nonTh17_VS_AS_nonTh17.txt", quote=FALSE, sep="\t", row.names=FALSE)
# ===============================================
res <- nbinomTest(cds, "AS_Th17", "AS_nonTh17")
write.table(res, file="diff_expression_AS_Th17_VS_AS_nonTh17.txt", quote=FALSE, sep="\t", row.names=FALSE)
# ===============================================
conds <- factor(c("HC", "HC", "HC", "HC", "HC", "HC", "HC", "AS", "AS", "AS", "AS", "AS", "AS"))
cds <- newCountDataSet(countsTable, conds)
cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)
res <- nbinomTest(cds, "HC", "AS")
write.table(res, file="diff_expression_HC_VS_AS.pooled.txt", quote=FALSE, sep="\t", row.names=FALSE)
```
## Differential expression analysis on high counts only

| Sample | Number of reads mapped to miRNAs |
| ------ | -------------------------------- |
| HC1-Th17    |    86,821 |
| HC2-nonTh17 |   995,035 |
| HC2-Th17    |   955,892 |
| HC3-nonTh17 |   677,173 |
| HC3-Th17    |   926,892 |
| HC4-nonTh17 |   450,159 |
| HC4-Th17    |   153,514 |
| AS1-nonTh17 | 1,188,284 |
| AS1-Th17    |   574,407 |
| AS2-nonTh17 |   298,505 |
| AS2-Th17    |    10,110 |
| AS3-nonTh17	|   665,606 |
| AS4-nonTh17 | 1,305,340 |

Excluding samples *HC1-Th17* and *AS2-Th17* from the
analysis, as they contains too few reads. Unfortunately
this analysis also did not produce any significant
difference in microRNA expression across conditions.


#### Designed by Irina Pulyakhina irina@well.ox.ac.uk
